### PR TITLE
Add Herringbone keyboard

### PR DESCRIPTION
**Pull request by ramonimbao**
## Description

Add Herringbone keyboard with VIA support

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/herringbone/readme.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
+# Herringbone
+
+![Herringbone](https://i.imgur.com/SQyS0j8.png)
+
+A short description of the keyboard/project
```
Comment (zvecr) [inline keyboards/herringbone/readme.md:5]: This should be filled in

File: keyboards/herringbone/keymaps/via/rules.mk [hunk 2]
```diff
@@ -0,0 +1,4 @@
+VIA_ENABLE = yes
+BOOTMAGIC_ENABLE = lite
```
Comment (zvecr) [inline keyboards/herringbone/keymaps/via/rules.mk:2]:
```suggestion
```

File: keyboards/herringbone/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    [0] = LAYOUT_ansi(
+        KC_ESC, KC_F1, KC_F2, KC_F3, KC_F4, KC_F5, KC_F6, KC_F7, KC_F8, KC_F9, KC_F10, KC_F11, KC_F12, KC_PSCR, KC_DEL,
+        KC_TILD, KC_1, KC_2, KC_3, KC_4, KC_5, KC_6, KC_7, KC_8, KC_9, KC_0, KC_MINS, KC_EQL,      KC_BSPC,     KC_HOME,
+        KC_TAB,    KC_Q, KC_W, KC_E, KC_R, KC_T, KC_Y, KC_U, KC_I, KC_O, KC_P, KC_LBRC, KC_RBRC, KC_BSLS,       KC_PGUP,
+        KC_CAPS,    KC_A, KC_S, KC_D, KC_F, KC_G, KC_H, KC_J, KC_K, KC_L, KC_SCLN, KC_QUOT,        KC_ENT,      KC_PGDN,
+        KC_LSFT,      KC_Z, KC_X, KC_C, KC_V, KC_B, KC_N, KC_M, KC_COMM, KC_DOT, KC_SLSH,    KC_RSFT,   KC_UP,  KC_END,
+        KC_LCTL,  KC_LGUI,  KC_LALT,              KC_SPC,              KC_RALT,  KC_RCTL,     KC_LEFT, KC_DOWN, KC_RIGHT
+    ),
```
Comment (zvecr) [inline keyboards/herringbone/keymaps/via/keymap.c:26]: The convention is that via keymaps init all the layers, generally 4, to `KC_TRANS`

File: keyboards/herringbone/rules.mk [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```
Comment (zvecr) [inline keyboards/herringbone/rules.mk:38]:
```suggestion
```

File: keyboards/herringbone/config.h [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+#define LCD_DATA3_PIN    7            //< pin for 4bit data bit 3
+#define LCD_RS_PORT      LCD_PORT     //< port for RS line
+#define LCD_RS_PIN       3            //< pin  for RS line
+#define LCD_RW_PORT      LCD_PORT     //< port for RW line
+#define LCD_RW_PIN       2            //< pin  for RW line
+#define LCD_E_PORT       LCD_PORT     //< port for Enable line
+#define LCD_E_PIN        1            //< pin  for Enable line
+#endif
+*/
+
```
Comment (fauxpark) [inline keyboards/herringbone/config.h:248]:
```suggestion
```

File: keyboards/herringbone/rules.mk [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+#   Atmel DFU    atmel-dfu
+#   LUFA DFU     lufa-dfu
+#   QMK DFU      qmk-dfu
+#   ATmega32A    bootloadHID
+#   ATmega328P   USBasp
+BOOTLOADER = USBasp
+
+# disable debug code
+OPT_DEFS = -DDEBUG_LEVEL=0
+
```
Comment (fauxpark) [inline keyboards/herringbone/rules.mk:19]:
```suggestion
```

**Comment by ramonimbao** [comment]
Thanks! I've made the changes. Some sections like the ones found in `herringbone.c` and the LCD were automatically generated with `new_keyboard.sh` so I thought not to remove them.

File: keyboards/herringbone/info.json [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+        }
+    }
+}
```
Comment (fauxpark) [inline keyboards/herringbone/info.json:18]:
```suggestion
{
    "keyboard_name": "Herringbone",
    "url": "",
    "maintainer": "Ramon Imbao",
    "width": 16.5,
    "height": 6.5,
    "layouts": {
        "LAYOUT_ansi": {
            "layout": [
                {"x": 0, "y": 0},

                {"x": 1.5, "y": 0},
                {"x": 2.5, "y": 0},
                {"x": 3.5, "y": 0},
                {"x": 4.5, "y": 0},

                {"x": 5.75, "y": 0},
                {"x": 6.75, "y": 0},
                {"x": 7.75, "y": 0},
                {"x": 8.75, "y": 0},

                {"x": 10, "y": 0},
                {"x": 11, "y": 0},
                {"x": 12, "y": 0},
                {"x": 13, "y": 0},

                {"x": 14.25, "y": 0},

                {"x": 15.5, "y": 0},

                {"x": 0, "y": 1.25},
                {"x": 1, "y": 1.25},
                {"x": 2, "y": 1.25},
                {"x": 3, "y": 1.25},
                {"x": 4, "y": 1.25},
                {"x": 5, "y": 1.25},
                {"x": 6, "y": 1.25},
                {"x": 7, "y": 1.25},
                {"x": 8, "y": 1.25},
                {"x": 9, "y": 1.25},
                {"x": 10, "y": 1.25},
                {"x": 11, "y": 1.25},
                {"x": 12, "y": 1.25},
                {"x": 13, "y": 1.25, "w": 2},

                {"x": 15.5, "y": 1.25},

                {"x": 0, "y": 2.25, "w": 1.5},
                {"x": 1.5, "y": 2.25},
                {"x": 2.5, "y": 2.25},
                {"x": 3.5, "y": 2.25},
                {"x": 4.5, "y": 2.25},
                {"x": 5.5, "y": 2.25},
                {"x": 6.5, "y": 2.25},
                {"x": 7.5, "y": 2.25},
                {"x": 8.5, "y": 2.25},
                {"x": 9.5, "y": 2.25},
                {"x": 10.5, "y": 2.25},
                {"x": 11.5, "y": 2.25},
                {"x": 12.5, "y": 2.25},
                {"x": 13.5, "y": 2.25, "w": 1.5},

                {"x": 15.5, "y": 2.25},

                {"x": 0, "y": 3.25, "w": 1.75},
                {"x": 1.75, "y": 3.25},
                {"x": 2.75, "y": 3.25},
                {"x": 3.75, "y": 3.25},
                {"x": 4.75, "y": 3.25},
                {"x": 5.75, "y": 3.25},
                {"x": 6.75, "y": 3.25},
                {"x": 7.75, "y": 3.25},
                {"x": 8.75, "y": 3.25},
                {"x": 9.75, "y": 3.25},
                {"x": 10.75, "y": 3.25},
                {"x": 11.75, "y": 3.25},
                {"x": 12.75, "y": 3.25, "w": 2.25},

                {"x": 15.5, "y": 3.25},

                {"x": 0, "y": 4.25, "w": 2.25},
                {"x": 2.25, "y": 4.25},
                {"x": 3.25, "y": 4.25},
                {"x": 4.25, "y": 4.25},
                {"x": 5.25, "y": 4.25},
                {"x": 6.25, "y": 4.25},
                {"x": 7.25, "y": 4.25},
                {"x": 8.25, "y": 4.25},
                {"x": 9.25, "y": 4.25},
                {"x": 10.25, "y": 4.25},
                {"x": 11.25, "y": 4.25},
                {"x": 12.25, "y": 4.25, "w": 1.75},

                {"x": 14.25, "y": 4.5},

                {"x": 15.5, "y": 4.25},

                {"x": 0, "y": 5.25, "w": 1.25},
                {"x": 1.25, "y": 5.25, "w": 1.25},
                {"x": 2.5, "y": 5.25, "w": 1.25},
                {"x": 3.75, "y": 5.25, "w": 6.25},
                {"x": 10, "y": 5.25, "w": 1.5},
                {"x": 11.5, "y": 5.25, "w": 1.5},

                {"x": 13.25, "y": 5.5},
                {"x": 14.25, "y": 5.5},
                {"x": 15.25, "y": 5.5}
            ]
        },
        "LAYOUT_iso": {
            "layout": [
                {"x": 0, "y": 0},

                {"x": 1.5, "y": 0},
                {"x": 2.5, "y": 0},
                {"x": 3.5, "y": 0},
                {"x": 4.5, "y": 0},

                {"x": 5.75, "y": 0},
                {"x": 6.75, "y": 0},
                {"x": 7.75, "y": 0},
                {"x": 8.75, "y": 0},

                {"x": 10, "y": 0},
                {"x": 11, "y": 0},
                {"x": 12, "y": 0},
                {"x": 13, "y": 0},

                {"x": 14.25, "y": 0},

                {"x": 15.5, "y": 0},

                {"x": 0, "y": 1.25},
                {"x": 1, "y": 1.25},
                {"x": 2, "y": 1.25},
                {"x": 3, "y": 1.25},
                {"x": 4, "y": 1.25},
                {"x": 5, "y": 1.25},
                {"x": 6, "y": 1.25},
                {"x": 7, "y": 1.25},
                {"x": 8, "y": 1.25},
                {"x": 9, "y": 1.25},
                {"x": 10, "y": 1.25},
                {"x": 11, "y": 1.25},
                {"x": 12, "y": 1.25},
                {"x": 13, "y": 1.25, "w": 2},

                {"x": 15.5, "y": 1.25},

                {"x": 0, "y": 2.25, "w": 1.5},
                {"x": 1.5, "y": 2.25},
                {"x": 2.5, "y": 2.25},
                {"x": 3.5, "y": 2.25},
                {"x": 4.5, "y": 2.25},
                {"x": 5.5, "y": 2.25},
                {"x": 6.5, "y": 2.25},
                {"x": 7.5, "y": 2.25},
                {"x": 8.5, "y": 2.25},
                {"x": 9.5, "y": 2.25},
                {"x": 10.5, "y": 2.25},
                {"x": 11.5, "y": 2.25},
                {"x": 12.5, "y": 2.25},
                {"x": 13.75, "y": 2.25, "w": 1.25, "h": 2},

                {"x": 15.5, "y": 2.25},

                {"x": 0, "y": 3.25, "w": 1.75},
                {"x": 1.75, "y": 3.25},
                {"x": 2.75, "y": 3.25},
                {"x": 3.75, "y": 3.25},
                {"x": 4.75, "y": 3.25},
                {"x": 5.75, "y": 3.25},
                {"x": 6.75, "y": 3.25},
                {"x": 7.75, "y": 3.25},
                {"x": 8.75, "y": 3.25},
                {"x": 9.75, "y": 3.25},
                {"x": 10.75, "y": 3.25},
                {"x": 11.75, "y": 3.25},
                {"x": 12.75, "y": 3.25},
                {"x": 15.5, "y": 3.25},

                {"x": 0, "y": 4.25, "w": 1.25},
                {"x": 1.25, "y": 4.25},
                {"x": 2.25, "y": 4.25},
                {"x": 3.25, "y": 4.25},
                {"x": 4.25, "y": 4.25},
                {"x": 5.25, "y": 4.25},
                {"x": 6.25, "y": 4.25},
                {"x": 7.25, "y": 4.25},
                {"x": 8.25, "y": 4.25},
                {"x": 9.25, "y": 4.25},
                {"x": 10.25, "y": 4.25},
                {"x": 11.25, "y": 4.25},
                {"x": 12.25, "y": 4.25, "w": 1.75},

                {"x": 14.25, "y": 4.5},

                {"x": 15.5, "y": 4.25},

                {"x": 0, "y": 5.25, "w": 1.25},
                {"x": 1.25, "y": 5.25, "w": 1.25},
                {"x": 2.5, "y": 5.25, "w": 1.25},
                {"x": 3.75, "y": 5.25, "w": 6.25},
                {"x": 10, "y": 5.25, "w": 1.5},
                {"x": 11.5, "y": 5.25, "w": 1.5},

                {"x": 13.25, "y": 5.5},
                {"x": 14.25, "y": 5.5},
                {"x": 15.25, "y": 5.5}
            ]
        },
        "LAYOUT_all": {
            "layout": [
                {"x": 0, "y": 0},

                {"x": 1.5, "y": 0},
                {"x": 2.5, "y": 0},
                {"x": 3.5, "y": 0},
                {"x": 4.5, "y": 0},

                {"x": 5.75, "y": 0},
                {"x": 6.75, "y": 0},
                {"x": 7.75, "y": 0},
                {"x": 8.75, "y": 0},

                {"x": 10, "y": 0},
                {"x": 11, "y": 0},
                {"x": 12, "y": 0},
                {"x": 13, "y": 0},

                {"x": 14.25, "y": 0},

                {"x": 15.5, "y": 0},

                {"x": 0, "y": 1.25},
                {"x": 1, "y": 1.25},
                {"x": 2, "y": 1.25},
                {"x": 3, "y": 1.25},
                {"x": 4, "y": 1.25},
                {"x": 5, "y": 1.25},
                {"x": 6, "y": 1.25},
                {"x": 7, "y": 1.25},
                {"x": 8, "y": 1.25},
                {"x": 9, "y": 1.25},
                {"x": 10, "y": 1.25},
                {"x": 11, "y": 1.25},
                {"x": 12, "y": 1.25},
                {"x": 13, "y": 1.25},
                {"x": 14, "y": 1.25},

                {"x": 15.5, "y": 1.25},

                {"x": 0, "y": 2.25, "w": 1.5},
                {"x": 1.5, "y": 2.25},
                {"x": 2.5, "y": 2.25},
                {"x": 3.5, "y": 2.25},
                {"x": 4.5, "y": 2.25},
                {"x": 5.5, "y": 2.25},
                {"x": 6.5, "y": 2.25},
                {"x": 7.5, "y": 2.25},
                {"x": 8.5, "y": 2.25},
                {"x": 9.5, "y": 2.25},
                {"x": 10.5, "y": 2.25},
                {"x": 11.5, "y": 2.25},
                {"x": 12.5, "y": 2.25},
                {"x": 13.5, "y": 2.25, "w": 1.5},

                {"x": 15.5, "y": 2.25},

                {"x": 0, "y": 3.25, "w": 1.75},
                {"x": 1.75, "y": 3.25},
                {"x": 2.75, "y": 3.25},
                {"x": 3.75, "y": 3.25},
                {"x": 4.75, "y": 3.25},
                {"x": 5.75, "y": 3.25},
                {"x": 6.75, "y": 3.25},
                {"x": 7.75, "y": 3.25},
                {"x": 8.75, "y": 3.25},
                {"x": 9.75, "y": 3.25},
                {"x": 10.75, "y": 3.25},
                {"x": 11.75, "y": 3.25},
                {"x": 12.75, "y": 3.25, "w": 2.25},

                {"x": 15.5, "y": 3.25},

                {"x": 0, "y": 4.25, "w": 1.25},
                {"x": 1.25, "y": 4.25},
                {"x": 2.25, "y": 4.25},
                {"x": 3.25, "y": 4.25},
                {"x": 4.25, "y": 4.25},
                {"x": 5.25, "y": 4.25},
                {"x": 6.25, "y": 4.25},
                {"x": 7.25, "y": 4.25},
                {"x": 8.25, "y": 4.25},
                {"x": 9.25, "y": 4.25},
                {"x": 10.25, "y": 4.25},
                {"x": 11.25, "y": 4.25},
                {"x": 12.25, "y": 4.25, "w": 1.75},

                {"x": 14.25, "y": 4.5},

                {"x": 15.5, "y": 4.25},

                {"x": 0, "y": 5.25, "w": 1.25},
                {"x": 1.25, "y": 5.25, "w": 1.25},
                {"x": 2.5, "y": 5.25, "w": 1.25},
                {"x": 3.75, "y": 5.25, "w": 6.25},
                {"x": 10, "y": 5.25, "w": 1.5},
                {"x": 11.5, "y": 5.25, "w": 1.5},

                {"x": 13.25, "y": 5.5},
                {"x": 14.25, "y": 5.5},
                {"x": 15.25, "y": 5.5}
            ]
        }
    }
}
```

Comment (ramonimbao) [reply keyboards/herringbone/info.json:18]: Wow, thanks for this and the other pull request! Truly!

File: keyboards/herringbone/config.h [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xB16B
```
Comment (Erovia) [inline keyboards/herringbone/config.h:23]: [Another keyboard](https://github.com/qmk/qmk_firmware/tree/master/keyboards/wete) already uses this VID.  
That board also has VIA support, so another VID will be needed.

You can check the currently used VID/PID values either [here](https://yanfali.github.io/qmk_usb_usage/) or with `git grep`.

File: keyboards/herringbone/keymaps/iso/rules.mk [hunk 9]
```diff
@@ -0,0 +1,3 @@
+VIA_ENABLE = yes
```
Comment (Erovia) [inline keyboards/herringbone/keymaps/iso/rules.mk:1]: I guess this file was supposed to be `keyboards/herringbone/keymaps/via/rules.mk`?

File: keyboards/herringbone/config.h (hunk 8 above)
Comment (ramonimbao) [reply keyboards/herringbone/config.h:23]: This particular keyboard is mine too lol. Alright, I thought I could get away with having the same VID but different PID.

Comment (Erovia) [reply keyboards/herringbone/config.h]: Oh, that's a crucial info I did not know, sorry about that.  
In this case the same VID would make sense.  

If you are planning on creating and contributing other boards (which I hope you do), it would also make sense to create a "vendor" directory and place your boards under it.  
This is done the sooner, the better, because for already existing boards it's a breaking change.

tl;dr: You should use the same VID for all your stuff and maybe move this to `keyboards/ramonimbao/herringbone`.

You `wete` board should also be moved there, but as I said, it's going to be a breaking change.

Comment (ramonimbao) [reply keyboards/herringbone/config.h]: I'm only about to fulfill the boards I've made so far in the local GB I did, but I haven't shipped them out yet. Will this be not be considered a breaking change anymore in that case? The only ones that have the PCBs so far are me and another local guy.

Comment (Erovia) [reply keyboards/herringbone/config.h]: I think this can be an exception, but the others will weigh in on it.  
I recommend doing the move as soon as possible in this case.

**Comment by Erovia** [comment]
Could you do that `git mv` here, too?
Currently this PR has both `keyboards/herringbone` and `keyboards/ramonimbao/herringbone`.

File: keyboards/ramonimbao/herringbone/config.h [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xB04E
```
Comment (Erovia) [inline keyboards/ramonimbao/herringbone/config.h:23]:
```suggestion
#define VENDOR_ID 0xB16B
```
So this goes back to the original one. :)

Comment (ramonimbao) [reply keyboards/ramonimbao/herringbone/config.h:23]: Right. haha
